pkgs/development/python-modules/octave-kernel/default.nix

```diff
@@ -0,0 +1,50 @@
+{
+  lib,
+  buildPythonPackage,
+  pythonOlder,
+  hatchling,
+  metakernel,
+  jupyter-client,
+  ipykernel,
+  nbconvert,
+  pytest,
+  fetchFromGitHub,
+}:
+
+buildPythonPackage rec {
+  pname = "octave_kernel";
+  version = "0.36.0";
+  pyproject = true;
+
+  disabled = pythonOlder "3.7";
+
+  src = fetchFromGitHub {
+    owner = "Calysto";
+    repo = "octave_kernel";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-/EmKUWCqlFlpJugIx7PLL99KJ5Lhyb/ropmfe9Symlw=";
+  };
+
+  build-system = [ hatchling ];
+
+  dependencies = [
+    metakernel
+    jupyter-client
+    ipykernel
+  ];
+
+  optional-dependencies = {
+    test = [
+      pytest
+      nbconvert
+    ];
```

Comment:
Please add `nativeCheckInputs = [ pytestCheckHook nbconvert ];` to ensure the tests are run.

And maybe remove `optional-dependencies.test`. Unless there is a reason to keep it.